velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -195,19 +195,36 @@ std::shared_ptr<const ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
     } else {
       if (schemaElement.repetition_type ==
           thrift::FieldRepetitionType::REPEATED) {
-        // child of LIST: "bag"
-        assert(children.size() == 1);
-        auto childrenCopy = children;
-        return std::make_shared<ParquetTypeWithId>(
-            TypeFactory<TypeKind::ARRAY>::create(children[0]->type),
-            std::move(childrenCopy),
-            curSchemaIdx,
-            maxSchemaElementIdx,
-            ParquetTypeWithId::kNonLeaf, // columnIdx,
-            schemaElement.name,
-            std::nullopt,
-            maxRepeat,
-            maxDefine);
+        VELOX_CHECK_LE(
+            children.size(), 2, "children size should not be larger than 2");
+        if (children.size() == 1) {
+          // child of LIST
+          auto childrenCopy = children;
+          return std::make_shared<ParquetTypeWithId>(
+              TypeFactory<TypeKind::ARRAY>::create(children[0]->type),
+              std::move(childrenCopy),
+              curSchemaIdx,
+              maxSchemaElementIdx,
+              ParquetTypeWithId::kNonLeaf, // columnIdx,
+              schemaElement.name,
+              std::nullopt,
+              maxRepeat,
+              maxDefine);
+        } else if (children.size() == 2) {
```

Comment:
Are you sure this is needed? The handling of map type should have been done in line 172 to 188. Could you provide a Parquet file that needs this change?

Reply:
@liushengxuan Have you confirmed if this is needed?